client/src/features/form/FormMoverControl.js

```diff
@@ -1,23 +1,127 @@
+
+  function checkForEmptyField(input) {
+    let fieldName = `${input}`;
+    let otherNameIGuess = "";
+
+    // if page 2
+    if (currentStep === 2) {
+      // Take the last four letters of input, call that the second word
+      otherNameIGuess += fieldName.slice(fieldName.length - 4, fieldName.length);
+      // the first word is either start or end, just see if the word includes it
+      if (fieldName.includes("start") || fieldName.includes("initial")) {
+        otherNameIGuess =  "Start " + otherNameIGuess;
+      } else {
+        otherNameIGuess =  "End " + otherNameIGuess;
+      }
+    }
+    // else, just set $input
+    else {
+      otherNameIGuess = fieldName;
+    }
+    if (formData[input] === "" || formData[input] == null) {
+      errorArray.push("Error: "+otherNameIGuess+" field can't be empty");
+    }
+  }
+
+  const handleNextButton = () => {
+
+    // Run a bunch of tests for each form page.
+    // For each test that returns an error, save a value.
+    // Return the values to the form page"s component.
+
+
+    setErrorArray([]);
+
+
+    switch (currentStep) {
+      case 1:
+
+
+        // Check if any values are empty string, null, undefined, and/or less than 2 characters
+        // Side note: This is the only case where you"d ever want to use == instead of ===
+
+        checkForEmptyField("title");
+        checkForEmptyField("description");
+        checkForEmptyField("location");
+
+
+        setFormCreateEventErrors(errorArray);
+        break;
+
+      // 2nd page - FormScheduleEvent
+      case 2:
+
+        // To start, if the event is not recurring, set startDate equal to endDate
+        if (formData["recurring"]["rate"] === "noRecurr") {
+          formData["finalDate"] = formData["initialDate"];
+        }
+
+        // Empty Field Value Tests
+        checkForEmptyField("initialDate");
+        checkForEmptyField("finalDate");
+        checkForEmptyField("startTime");
+        checkForEmptyField("endTime");
+
+        // Start Date & End Date Cannot be more than 90 days apart
+        // get date 90 days before final date
+        const NinetyDaysBeforeFinalDate = sub(parseISO(formData["finalDate"]), { days: 90 });
+        if (parseISO(formData["initialDate"]) < NinetyDaysBeforeFinalDate) {
+          errorArray.push("Error: Start date and End date cannot be more than 90 days apart");
+        }
+
+        // Start Date cannot be after End Date, vica versa
+        if (parseISO(formData["finalDate"]) < parseISO(formData["initialDate"])) {
+          errorArray.push("Error: Start date cannot be after End date");
+        }
+
+        // "Weekly" Recurring Event MUST include at least on day of week
+        if (formData["recurring"]["rate"] === "weekly" && formData["recurring"]["days"].length === 0) {
+          errorArray.push("Error: Weekly recurring event MUST include at least one day of the week");
+        }
+
+        // Start time cannot be after End time
```

Comment:
This can be an issue if the event starts before midnight and ends after midnight. For example, Leon's classes last from 11:30 PM to 2:30 AM if you're in Europe.